src/modal/modal.js

```diff
@@ -106,9 +106,31 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap'])
     }
   }])
 
+/**
+ * A helper, for measuring and maintaining scrollbar properties - used by modalWindow and $modalStack
+ */
+  .factory('$$scrollbarHelper', function () {
+    var scrollbarWidth;
+
+    return {
+      measureScrollbar: measureScrollbar,
+      scrollbarWidth: scrollbarWidth
+    };
+
+    // from modal.js of bootstrap
+    function measureScrollbar () { // thx walsh
+      var scrollDiv = document.createElement('div');
+      scrollDiv.className = 'modal-scrollbar-measure';
+      document.body.appendChild(scrollDiv);
+      var width = scrollDiv.offsetWidth - scrollDiv.clientWidth;
+      document.body.removeChild(scrollDiv);
+      this.scrollbarWidth = width;
+    }
+  })
+
```

Comment:
I'd like to see this in the $position service as scrollbarWidth(), we might need to use it for auto/smart positioning. Would be nice to cache the result to avoid manipulating the DOM every time we need this value as it should not change.